mbbsd/vote.c

```diff
@@ -966,6 +1144,48 @@ user_vote_one(const vote_buffer_t *vbuf, const char *bname)
 		    }
 		}
 		move(b_lines - 1, 0);
+
+		/* log this vote */
+		if (logconf.log_date || logconf.log_id || logconf.log_choice || logconf.log_ip) {
+		    int fd_log, len;
+
+		    char genbuf[300], tmpbuf[64];
+		    int offset = 0;
+
+		    if (logconf.log_id && offset >= 0) {
+			snprintf(tmpbuf, 64, "�벼�H %-12s ", cuser.userid);
+			compose_vote_log_buffer(genbuf, sizeof(genbuf), &offset, tmpbuf);
```

Comment:
No need to reinvent this. Use `strlcat`. E.g.
`strlcat(genbuf, tmpbuf, sizeof(genbuf));`
Better, no need to check offset, when it's full, it's full.